### PR TITLE
added workflows, updated dependencies and replaced lazy_static with once_cell

### PR DESCRIPTION
**Pull request by sneedcat**

### CODE REVIEW

**Comment by oppiliappan** [comment]
This looks amazing, thanks for the work on this. I am a little busy at the moment to test, but I'll review this in a couple of days.

Thanks again!

File: src/main.rs [hunk 1]
```diff
@@ -37,19 +37,15 @@ struct Configuration {
 }
 
 #[cfg(not(test))]
-lazy_static! {
-    static ref CONFIGURATION: Configuration = parse_arguments();
-}
+static CONFIGURATION: Lazy<Configuration> = Lazy::new(|| parse_arguments());
```
Comment (pickfire) [inline src/main.rs]:
```suggestion
static CONFIGURATION: Lazy<Configuration> = Lazy::new(parse_arguments);
```
Should we do this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        - name: macos-build
+          os: macos-latest
+
+        - name: windows-build
+          os: windows-latest
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
```
Comment (pickfire) [inline .github/workflows/ci.yml:26]: Why not run rustfmt and clippy as well?

**Comment by sneedcat** [comment]
I made the suggested changes and added `cargo test` and `cargo clippy` to workflow.

**Review by pickfire — APPROVED** [review]
Other than a style comment looks good to me.

File: src/lex.rs [hunk 3]
```diff
@@ -67,59 +67,110 @@ pub enum Token {
...
+    m.insert(
+        "sin",
+        Function::token_from_fn("sin".into(), |x| {
+            is_radian_mode(x, CONFIGURATION.radian_mode).sin()
+        }),
+    );
+    m.insert(
+        "cos",
+        Function::token_from_fn("cos".into(), |x| {
+            is_radian_mode(x, CONFIGURATION.radian_mode).cos()
```
Comment (pickfire) [inline src/lex.rs:88]: I think it might be better to clippy ignore this part, the older part seemed more readable.

**Comment by pickfire** [comment]
@meguminloli Thanks a lot for working on this, but unfortunately it is outdated, let me take your changes and merge it. Sorry for the late reply.

**Comment by pickfire** [comment]
The changes are now in master, will be closing this.

**Pull request closed without merging**